### PR TITLE
v0.5.0: pkg: Pin to RHCOS 47.167 and quay.io/openshift-release-dev/ocp-release:4.0.0-3

### PR DESCRIPTION
**Pull request by wking**
DO NOT MERGE!

That's the latest RHCOS release:

```console
$ curl -s https://releases-rhcos.svc.ci.openshift.org/storage/releases/maipo/builds.json | jq '{latest: .builds[0], timestamp}'
{
  "latest": "47.165",
  "timestamp": "2018-12-02T06:41:22Z"
}
```

And @smarterclayton just [pushed 4.0.0-0.alpha-2018-12-02-020136 to quay.io/openshift-release-dev/ocp-release:4.0.0-3][1].

This PR builds on #772.  The CHANGELOG commit (also in #772) will be merged into master.  The pin commit (unique to this PR) will get the v0.5.0 tag, but not be merged into master.  I'm just targetting master with the PR because I want CI to pass on the pinned dependencies before we tag the release.

/hold

[1]: https://github.com/openshift/installer/pull/772#issuecomment-443486056

### CODE REVIEW

**Comment by wking** [comment]
In case a graph is easier to understand, here's where I think we want to be post-release:

```
*   9a6ae15 (tag: v0.5.0-master, origin/master) Merge branch 'version-0.5.0'
|\  
| | * 54ba349 (tag: v0.5.0, origin/pr/773) pkg: Pin to RHCOS 47.167 and quay.io/openshift-release-dev/ocp-release:4.0.0-3
| |/  
| * fc0e947 (origin/pr/772) CHANGELOG: Document changes since v0.4.0
|/  
*   12ef0cb Merge pull request #748 from ingvagabund/change-machine-apiversion
```

Modulo any hash changes due to future rebasing of #772 or #773, and Tide generating the merge that is `9a6ae15` in that graph instead of me generating it locally.

**Comment by wking** [comment]
[e2e-aws][1] included:

```
fail [github.com/openshift/origin/test/extended/operators/operators.go:78]: Dec 2 10:01:08.686: ClusterVersion never became available: Unable to apply 4.0.0-3: a required extension is not available to update
```

Maybe a flake?  Or due to pulling the payload from quay.io, @smarterclayton?

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/773/pull-ci-openshift-installer-master-e2e-aws/1859/artifacts/e2e-aws/junit/junit_2018-12-02T102046.xml

**Comment by smarterclayton** [comment]
Crap.  That’s the but with cvo processing files.  I’ll have to run the older release command (so this won’t be a valid Cincinnati release).

Will push with an older binary shortly

**Comment by smarterclayton** [comment]
Bug

**Comment by smarterclayton** [comment]
I have to remirror, there's a bug in release publishing that didn't update
the references to the new location during a mirror.

**Comment by wking** [comment]
[e2e-aws][1] timed out waiting for `bootstrap-complete`:

```
level=warning msg="Failed to connect events watcher: Get https://ci-op-5g5vrcw1-1d3f3-api.origin-ci-int-aws.dev.rhcloud.com:6443/api/v1/namespaces/kube-system/events?watch=true: dial tcp 52.2.108.126:6443: connect: connection refused"
level=fatal msg="Error executing openshift-install: waiting for bootstrap-complete: timed out waiting for the condition"
```

I'm going to launch it again and see if I can grab the error from the CI nodes before they get torn down.

/retest

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/773/pull-ci-openshift-installer-master-e2e-aws/1861/build-log.txt

**Comment by wking** [comment]
From [this run][1]'s bootstrap node:

```console
[core@ip-10-0-10-189 ~]$ journalctl -n5 -u bootkube.service | cat
-- Logs begin at Mon 2018-12-03 06:54:37 UTC, end at Mon 2018-12-03 07:24:52 UTC. --
Dec 03 07:24:48 ip-10-0-10-189 bootkube.sh[696]: Trying to pull quay.io/openshift-release-dev/ocp-v4.0@sha256:69bae91f6a933045175170cebd1caf00b8216a096fd4d402c41957a99aa3435b...Failed
Dec 03 07:24:48 ip-10-0-10-189 bootkube.sh[696]: unable to pull quay.io/openshift-release-dev/ocp-v4.0@sha256:69bae91f6a933045175170cebd1caf00b8216a096fd4d402c41957a99aa3435b: unable to pull image: Error determining manifest MIME type for docker://quay.io/openshift-release-dev/ocp-v4.0@sha256:69bae91f6a933045175170cebd1caf00b8216a096fd4d402c41957a99aa3435b: Error reading manifest sha256:69bae91f6a933045175170cebd1caf00b8216a096fd4d402c41957a99aa3435b in quay.io/openshift-release-dev/ocp-v4.0: unauthorized: access to the requested resource is not authorized
Dec 03 07:24:48 ip-10-0-10-189 systemd[1]: bootkube.service: main process exited, code=exited, status=125/n/a
Dec 03 07:24:48 ip-10-0-10-189 systemd[1]: Unit bootkube.service entered failed state.
Dec 03 07:24:48 ip-10-0-10-189 systemd[1]: bootkube.service failed.
```

The secret in the bootstrap's `/opt/tectonic/manifests/pull.json` has an entry for `quay.io`, and when I copy the secret's `.dockerconfigjson` down (and base64 decode it) to my dev box and run:

```console
$ podman pull --authfile path/to/ci-pull.json quay.io/openshift-release-dev/ocp-v4.0@sha256:69bae91f6a933045175170cebd1caf00b8216a096fd4d402c41957a99aa3435b
```

it works.  I'll add the pull JSON (without any Kubernetes wrapping) to the bootstrap ignition and point `REGISTRY_AUTH_FILE` at it.

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/773/pull-ci-openshift-installer-master-e2e-aws/1862/build-log.txt

**Comment by wking** [comment]
> I'll add the pull JSON (without any Kubernetes wrapping) to the bootstrap ignition and point `REGISTRY_AUTH_FILE` at it.

Done in #775.  I've cherry-picked that commit onto this branch so we can test it against protected content.  Once both #775 and this branch are green, we can land #775 and rebase #772 and #773 on top.

File: pkg/asset/ignition/bootstrap/bootstrap.go [hunk 1]
```diff
@@ -141,7 +142,7 @@ func (a *Bootstrap) getTemplateData(installConfig *types.InstallConfig, adminKub
 	}
 
 	releaseImage := defaultReleaseImage
-	if ri, ok := os.LookupEnv("OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE"); ok && ri != "" {
+	if ri, ok := os.LookupEnv("_OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE"); ok && ri != "" {
```
Comment (sjenning) [inline pkg/asset/ignition/bootstrap/bootstrap.go]: why?

Comment (wking) [reply pkg/asset/ignition/bootstrap/bootstrap.go]: Last paragraph from f8a93b33f72fbd10e02d9a8688c48b97d62c34f5

**Comment by wking** [comment]
Rebased onto the current #772 tip, so now both are on top of master since #775 landed, and we're back to a target release graph like [the original plan][1].

[1]: https://github.com/openshift/installer/pull/773#issuecomment-443491971

**Comment by wking** [comment]
[v0.5.0][1] cut off this PR.

/close

[1]: https://github.com/openshift/installer/releases/tag/v0.5.0

**Comment by wking** [comment]
/test images

@smarterclayton wanted fresh builds from this image

**Pull request closed without merging**